setup.py

```diff
@@ -1,8 +1,22 @@
+import sys
 from setuptools import setup
 
+
 with open('README.rst', 'rb') as f:
     readme = f.read().decode('utf-8')
 
+
+requires = [
+    'Click'
+]
+
+
+if sys.version_info.major == 2:
+    requires.extend([
+        'virtualenv'
```

Comment:
oh, that one is always required - else one would be dependent on the system one

Reply:
Hmm, but that's the purpose of this PR right? To use builtin package `venv` instead of `virtualenv` in Python 3. And since the code here https://github.com/mitsuhiko/pipsi/pull/142/files#diff-bc98b82c8d093ef76c369fe573d188cbR343 has already handled this situation, `virtualenv` is not event imported when using Python 3. So I think it could be directly removed from the requires. What do you think?

Reply:
@reorx creating a python 2 env will require virtualenv - so until python 2 support is dropped, its a required dependency

Reply:
I got you, but there maybe a bit confusing here, let's make it clear by enumerate different situations:
1. If `get-pipsi.py` is executed by python 2, then `virtualenv` will be included in `requires`.
    - `pipsi install foo`, this will use `~/.local/venvs/pipsi/bin/python` to create python 2 virtualenv for foo, works ok
   - `pipsi install --python /some/other/python2 foo`, in this situation whether virtualenv for foo can be created depends on `/some/other/python2` has `virtualenv` installed or not, nothing to do with pipsi requires `virtualenv` or not.
   - `pipsi install --python /some/python3 foo`, in this situation pipsi will call `/some/python3 -m venv` to create virtualenv for foo, works ok

2. If `get-pipsi.py` is executed by python 3, then `virtualenv` will not be included in `requires`.
    - `pipsi install foo`, this will call `~/.local/venvs/pipsi/bin/python -m venv` to create python 3 virtualenv for foo, works ok
   - `pipsi install --python /some/python2 foo`, in this situation whether virtualenv for foo can be created depends on `/some/python2` has `virtualenv` installed or not, nothing to do with pipsi requires `virtualenv` or not.
   - `pipsi install --python /some/other/python3 foo`, in this situation pipsi will call `/some/other/python3 -m venv` to create virtualenv for foo, works ok

I think these are all the situations pipsi can meet, each shows that "ignoring virtualenv for pipsi in Python 3" will not have bad influence on how pipsi works. What's your idea?

Reply:
Hey I hope I didn't get it wrong, by saying "requiring a external virtualenv for the pipsi use-case is not acceptable" did you mean pipsi should not use virtualenv if it's possible? Which is the same idea with mine.

Reply:
OK so I finally find out what's going on here, the main reason we are not speaking the same voice is because, I changed `args = [sys.executable, '-m', 'virtualenv', '-p', python or sys.executable, venv_path]` to `[python, '-m', 'virtualenv', venv_path]` previously, without much thoughts it became the subconscious when I was thinking about whether `virtualenv` is required or not, and that makes things all wrong.

The right way is to only use `venv` if target python is version 3, otherwise `virtualenv` should always be used to create for python 2. Whether pipsi is installed in python 2 or 3 makes no difference here (before realized this, I thought a `virtualenv` installed in python 2 can not create python 3 virtualenv, vice versa).

I've updated the PR with new commits, please help to see if I do it right this time.